algorithms/src/crh/bhp.rs

```diff
@@ -96,164 +111,34 @@ impl<G: ProjectiveCurve, const NUM_WINDOWS: usize, const WINDOW_SIZE: usize> CRH
 }
 
 impl<G: ProjectiveCurve, const NUM_WINDOWS: usize, const WINDOW_SIZE: usize> BHPCRH<G, NUM_WINDOWS, WINDOW_SIZE> {
-    pub fn create_generators(message: &str) -> Vec<Vec<G>> {
-        let mut generators = Vec::with_capacity(NUM_WINDOWS);
-        for index in 0..NUM_WINDOWS {
-            // Construct an indexed message to attempt to sample a base.
-            let indexed_message = format!("{} at {}", message, index);
-            let (generator, _, _) = hash_to_curve::<G::Affine>(&indexed_message);
-            let mut base = generator.into_projective();
-            // Compute the generators for the sampled base.
-            let mut generators_for_segment = Vec::with_capacity(WINDOW_SIZE);
-            for _ in 0..WINDOW_SIZE {
-                generators_for_segment.push(base);
-                for _ in 0..4 {
-                    base.double_in_place();
-                }
-            }
-            generators.push(generators_for_segment);
-        }
-        generators
-    }
-
-    pub fn base_lookup(&self, bases: &[Vec<G>]) -> &Vec<Vec<[G; BOWE_HOPWOOD_LOOKUP_SIZE]>> {
-        self.base_lookup
-            .get_or_try_init::<_, ()>(|| {
-                Ok(crate::cfg_iter!(bases)
-                    .map(|x| {
-                        x.iter()
-                            .map(|g| {
-                                let mut out = [G::zero(); BOWE_HOPWOOD_LOOKUP_SIZE];
-                                for (i, element) in out.iter_mut().enumerate().take(BOWE_HOPWOOD_LOOKUP_SIZE) {
-                                    let mut encoded = *g;
-                                    if (i & 0x01) != 0 {
-                                        encoded += g;
-                                    }
-                                    if (i & 0x02) != 0 {
-                                        encoded += g.double();
-                                    }
-                                    if (i & 0x04) != 0 {
-                                        encoded = encoded.neg();
-                                    }
-                                    *element = encoded;
-                                }
-                                out
-                            })
-                            .collect()
-                    })
-                    .collect())
-            })
-            .expect("failed to init BoweHopwoodPedersenCRHParameters")
-    }
-
-    /// Precondition: number of elements in `input` == `num_bits`.
-    pub(crate) fn hash_bits_inner<S: Borrow<bool>>(
-        &self,
-        input: impl Iterator<Item = S>,
-        num_bits: usize,
-    ) -> Result<G, CRHError> {
-        if num_bits > WINDOW_SIZE * NUM_WINDOWS {
-            return Err(CRHError::IncorrectInputLength(num_bits, WINDOW_SIZE, NUM_WINDOWS));
+    pub(crate) fn hash_bits_inner(&self, input: &[bool]) -> Result<G, CRHError> {
```

Comment:
This should just output `G::BaseField`.

Reply:
The BHP commitment needs the output to compute the randomness on top of it.

Reply:
ah good point.